### ISSUE TITLE
Check whether webbooost can be ported to Opera, Safari and FF

### ISSUE THREAD

**Issue by mr-mig** [mr-mig/webbooost#13]

**Comment by mubaidr** [comment]
Opera is based on Chromium, so there should very little required to port it to Opera (Or maybe nothing, as chrome version is working in Opera).

**Comment by mr-mig** [comment]
Unfortunately, it cannot be easily ported. 
Opera and Safari does not have the same API for extensions available (you cannot hijack requests from web pages).

Firefox is a bit convoluted and requires a lot of workarounds for this extension to be ported. But it is definitely possible.

**Comment by mubaidr** [comment]
I cannot say anything about Safari, But Opera does provide api for request blocking, I have written an extension myself. 

Here is detailed info for Opera API:
https://dev.opera.com/extensions/apis/

As i sadi before, Opera and Chrome are very very similar since they both are based on Chromium.

**Comment by mr-mig** [comment]
I was sure they have not yet reached feature parity for webRequest. 

I'll check it out, it would be really nice to have web boost available for Opera.

**Comment by mubaidr** [comment]
@mr-mig :+1:

**Comment by mr-mig** [comment]
@mubaidr it works out of the box.
Do you want to test the extension?

**Comment by mubaidr** [comment]
I am already using it (Installed from Chrome webstore), If you have new beta you can send me.

**Comment by mr-mig** [comment]
@mubaidr :+1:

**Comment by mubaidr** [comment]
Automatic updates will not work this way. So i hope you re-brand it for Opera Add-on store.

**Comment by mr-mig** [comment]
@mubaidr I've already uploaded it to the Opera web store. Waiting for a moderator review :)

**Comment by mubaidr** [comment]
Awesome!:+1:

**Comment by mubaidr** [comment]
@mr-mig Still not available in Opera store, any specific reason?

**Comment by mr-mig** [comment]
@mubaidr Yeah, still waiting for a review. It is manual, apparently :disappointed:

**Comment by mr-mig** [comment]
@mubaidr Here it is! https://addons.opera.com/en/extensions/details/web-boost-wait-less-browse-faster/

**Comment by cubakcabak** [comment]
sorry to ask, i'm power user not coder.

is there any progress for firefox? i've installed it on chrome and it tremendously help in fastening browsing experiences.

**Comment by mr-mig** [comment]
@cubakcabak unfortunately no, there is no progress. 
Web Boost depends on the capabilities, which are not in firefox yet. 
It will be added in the browser in the future, but right now there is no way to make it work reliably.
